envoy/config/filter/http/jwt_authn/v2/config.proto

```diff
@@ -0,0 +1,170 @@
+  //
+  // 1. The Authorization header using the Bearer schema.
+  //    See `here <https://tools.ietf.org/html/rfc6750#section-2.1>`_
+  //    Example::
+  //
+  //       Authorization: Bearer <token>.
+  //
+  // 2. `access_token` query parameter.
+  //    See `here <https://tools.ietf.org/html/rfc6750#section-2.3>`_
+  //
+  //
+  // Only the first JWT token is verified per request if multiple tokens are
+  // specified.
+  // If both jwt_headers and jwt_params are specified, jwt_headers will be
+  // checked first.
+  //
+  // JWT is sent in a request header. `jwt_headers` represents the
+  // header names. Their format is::
+  //
+  //    header_name: <JWT>
+  //
+  // For example, if config is::
+  //
+  //   jwt_headers:
+  //      x-goog-iap-jwt-assertion
+  //
+  // The header format will be::
+  //
+  //   x-goog-iap-jwt-assertion: <JWT>.
+  //
+  repeated string jwt_headers = 6;
+
+  // JWT is sent in a query parameter. `jwt_params` represents the
+  // query parameter names.
+  //
+  // For example, if config is::
+  //
+  //   jwt_params:
+  //      jwt_token
+  //
+  // The query parameter is::
+  //
+  //    /path?jwt_token=<JWT>
+  //
+  repeated string jwt_params = 8;
+
+  // This field is specific for Envoy proxy implementation.
+  // It is the cluster name in the Envoy "cluster_manager" config section.
+  // In order for Envoy to call "jwks_uri", its host has to be specified
+  // as a "cluster" in the config for each jwks_uri.
+  string jwks_uri_cluster = 7;
+}
+
+// This is the Envoy filter config for JSON Web Token authentication.
+// [#not-implemented-hide:]
+message JSONWebTokenAuthentication {
+  // List of JWT rules to valide.
+  //
+  // If the request includes a JWT, it must match one of the JWT listed
+  // here with the same issuer.
+  //
+  repeated JSONWebToken jwts = 2;
+
+  // This message defines a pattern to match a HTTP request.
+  // A pattern is matched only if all "headers" are matched.
+  message HttpMatcher {
+    // Specifies a set of headers should match on.
+    // For example::
+    //
+    //    headers:
+    //    - name: :method
+    //      value: OPTIONS
+    //    headers:
+    //    - name: :path
+    //      regex_match: /.*
+    //
+    repeated envoy.api.v2.route.HeaderMatcher headers = 6;
+  }
+
+  // If a request doesn't have JWT, it will be rejected.
+  // But some requests can be forwarded without JWT, such as OPTIONS for CORS
+  // and some health checking paths.
+  //
+  // Examples: bypass all CORS options requests::
+  //
+  //   bypass_jwt:
+  //      headers:
+  //      - name: :method
+  //        value: OPTIONS
+  //      headers:
+  //      - name: :path
+  //        regex_match: /.*
+  //
+  // Examples: bypass /healthz check::
+  //
+  //   bypass_jwt:
+  //      headers:
+  //      - name: :method
+  //        value: GET
+  //      headers:
+  //      - name: :path
+  //        exact_match: /healthz
+  //
+  repeated HttpMatcher bypass_jwt = 3;
+}
```

Comment:
You might need an action here (or in `JSONWebToken`) to define what the filter do when it cannot verify a JWT or a JWT doesn't exists.

Reply:
Done